extensions/notebook/src/book/bookModel.ts

```diff
@@ -56,14 +59,36 @@ export class BookModel {
 	}
 
 	public async initializeContents(): Promise<void> {
-		this._bookItems = [];
-		this._allNotebooks = new Map<string, BookTreeItem>();
-		if (this.isNotebook) {
-			this.readNotebook();
-		} else {
-			await this.readBookStructure();
-			await this.loadTableOfContentFiles();
-			await this.readBooks();
+		const deferred = new Deferred<void>();
+		deferred.then(async () => {
+			this._queuedPromises.shift()?.resolve();
+		});
+		if (!this._activePromise && this._queuedPromises.length === 0) {
+			this._activePromise = deferred;
+		}
+		else {
+			// If there's an active promise, then we need to add the new promise to the queue.
+			this._queuedPromises.push(deferred);
+			await deferred.promise;
+		}
+		try {
+			this._bookItems = [];
+			this._allNotebooks = new Map<string, BookTreeItem>();
+			if (this.isNotebook) {
+				this.readNotebook();
+			} else {
+				await this.readBookStructure();
+				await this.loadTableOfContentFiles();
+				await this.readBooks();
+			}
+		}
+		finally {
+			// Resolve promise after initializing book
+			this._activePromise.resolve();
```

Comment:
You're not resolving the active promise here - that's the one that was already resolved. Instead you need to pull off the first item in the queue and if there's one there resolve that so that it continues execution 

```
const queuedPromise = this._queuedPromises.shift();
if(queuedPromise) {
   queuedPromise.resolve();
}
this._activePromise = queuedPromise;
```